### PR TITLE
[12.0][MIG] Migrated sale_report_hide_price

### PR DESCRIPTION
**Pull request by Chanakya-SerpentCS**
[12.0][MIG] Migrated sale_report_hide_price

### CODE REVIEW

File: sale_report_hide_price/views/report_saleorder.xml [hunk 1]
```diff
@@ -61,7 +50,7 @@
                                 <td class="text-right">-N.A.-</td>
                             </t>
                         </tr>
-                        <t t-foreach="doc._get_tax_amount_by_group()" t-as="amount_by_group">
+                        <t t-foreach="doc.amount_by_group" t-as="amount_by_group">
```
Comment (JayVora-SerpentCS) [inline sale_report_hide_price/views/report_saleorder.xml:53]: explain each change done plz

File: sale_report_hide_price/views/report_saleorder.xml [hunk 2]
```diff
@@ -6,49 +6,38 @@
...
-                    <span t-field="l.discount" />
+                    <span t-field="line.discount" />
                 </td>
             </t>
             <t t-if="not doc.show_discount">
                 <td class="text-right" groups="sale.group_discount_per_so_line">-N.A.-</td>
             </t>
         </td>
-        <xpath expr="//table[hasclass('table-condensed')]/tbody/t/t[2]/tr/td[3]" position="replace">
+        <xpath expr="//table[hasclass('table-sm')]/tbody/t[2]/tr/t/td[3]" position="replace">
```
Comment (Chanakya-SerpentCS) [inline sale_report_hide_price/views/report_saleorder.xml:16]: @JayVora-SerpentCS  Path of  field line.price_unit is changed from v11 to v12.

File: sale_report_hide_price/views/report_saleorder.xml [hunk 3]
```diff
@@ -6,49 +6,38 @@
...
             <t t-if="doc.show_price">
                 <td class="text-right">
-                    <span t-field="l.price_unit" />
+                    <span t-field="line.price_unit" />
```
Comment (Chanakya-SerpentCS) [inline sale_report_hide_price/views/report_saleorder.xml:19]: Name of line object is changed from "l" to "line" from v11 to v12.

File: sale_report_hide_price/views/report_saleorder.xml [hunk 4]
```diff
@@ -6,49 +6,38 @@
...
         </xpath>
-        <xpath expr="//table[hasclass('table-condensed')]/tbody/t/t[2]/tr/td[last()-1]" position="replace">
+        <xpath expr="//table[hasclass('table-sm')]/tbody/t[2]/tr/t/td[last()]" position="replace">
             <t t-if="doc.show_price">
-                <td class="text-right" groups="sale.group_show_price_subtotal">
-                    <span t-field="l.price_subtotal"
-                          t-options='{"widget": "monetary", "display_currency": doc.pricelist_id.currency_id}' />
+                <td class="text-right" groups="account.group_show_line_subtotals_tax_excluded">
+                     <span t-field="line.price_subtotal" groups="account.group_show_line_subtotals_tax_excluded"/>
+                     <span t-field="line.price_total" groups="account.group_show_line_subtotals_tax_included"/>
```
Comment (Chanakya-SerpentCS) [inline sale_report_hide_price/views/report_saleorder.xml:30]: @JayVora-SerpentCS  in v12 Total Price(price_total) is included in same < td > where price_subtotal field is. Previously in v11  they were in different < td >.

File: sale_report_hide_price/views/report_saleorder.xml [hunk 5]
```diff
@@ -6,49 +6,38 @@
...
-                <td class="text-right" groups="sale.group_show_price_total">-N.A.-</td>
+                <td class="text-right" groups="account.group_show_line_subtotals_tax_excluded">-N.A.-</td>
             </t>
         </xpath>
         <xpath expr="//div[@name='total']" position="replace">
             <div class="row" name="total">
-                <div class="col-xs-4 pull-right">
-                    <table class="table table-condensed">
+                <div t-attf-class="#{'col-4' if report_type != 'html' else 'col-sm-7 col-md-5'} ml-auto">
+                    <table class="table table-sm">
```
Comment (Chanakya-SerpentCS) [inline sale_report_hide_price/views/report_saleorder.xml:40]: @JayVora-SerpentCS  This are the changes done in v12 Sales Order report which I have added in this inherited report to keep design related changes in v12 as it is .

File: sale_report_hide_price/views/report_saleorder.xml (hunk 1 above)
Comment (Chanakya-SerpentCS) [reply sale_report_hide_price/views/report_saleorder.xml:53]: @JayVora-SerpentCS  This method _get_tax_amount_by_group is no more in v12. it has changed with amount_by_group.

**Pull request merged**